src/persist/src/indexed/encoding.rs

```diff
@@ -132,8 +132,13 @@ pub struct BlobFutureBatch<K, V> {
 /// The structure serialized and stored as a value in [crate::storage::Blob]
 /// storage for data keys corresponding to trace data.
 ///
+/// Users iterating through `updates` must take care to advance times less than
```

Comment:
This doesn't really express what I want to express. More specifically, what I think is true is "you can only trust the batch to provide completely correct data for times at or in advance of since. If you're iterating through updates to say rehydrate a dataflow, you don't need to advance them to the since frontier. but the minute you want to ask a question like 'what was the state at time X' you need to be sure that X is at or in advance of the since frontier"

Reply:
this is all based on multiple readings of https://docs.rs/differential-dataflow/0.12.0/src/differential_dataflow/trace/description.rs.html (still not totally sure i understand it correctly)

Reply:
How about

> This batch represents the data that was originally written at some time in `[lower, upper)` (more precisely `!< lower` and `< upper`). The individual record times may have later been forwarded by compaction to something `<= since`. This means the ability to reconstruct the state of the collection at times `< since` has been lost.

Reply:
I like it! I pushed something basically identical with a more explicit warning that the batch may still contain records at times < since, at which point it is the reader's responsibility to advance times to since

Reply:
OOC when will there be times < since in the batch?

Reply:
right now -- never. I left this text in there to give a degree of freedom around updating the metadata vs doing the compaction that we may not actually need.